domain/schema/model/sql/0017-constraint.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE "constraint" (
+    uuid                TEXT PRIMARY KEY,
+    arch                TEXT,
+    cpu_cores           INT,
+    cpu_power           INT,
+    mem                 INT,
+    root_disk           INT,
+    root_disk_source    TEXT,
+    instance_role       TEXT,
+    instance_type       TEXT,
+    container_type_id   INT,
+    virt_type           TEXT,
+    allocate_public_ip  BOOLEAN,
+    image_id            TEXT,
+    CONSTRAINT          fk_constraint_container_type
+        FOREIGN KEY     (container_type_id)
+        REFERENCES      container_type(id)
+);
+
+CREATE TABLE constraint_tag (
```

Comment:
Primary keys for all of these need to be the constraint UUID + tag/space/zone, otherwise there can only be a single row per constraint.